jekyll/_cci2/writing-yaml.md

```diff
@@ -0,0 +1,166 @@
+
+This document gives an overview of YAML,
+a data serialization language used to configure CircleCI.
+
+## Overview
+
+[YAML](http://yaml.org) is a human-friendly data serialization standard for all programming languages.
+It is a strict superset of [JSON](https://www.json.org/),
+another data serialization language.
+This means it can do everything JSON can... and more.
+
+CircleCI configuration is stored in a single YAML file located at `~/.circleci/config.yml`,
+where `~` is the root of your project's directory.
+Since most of your work with CircleCI occurs in this file,
+it is important to understand the basics of YAML formatting.
+
+## How to Write YAML
+
+The basic structure of a YAML file is a [map](https://en.wikipedia.org/wiki/Map_(higher-order_function))
+and consists of one or more key-value pairs.
+
+```yaml
+key: value
+```
+
+You can set another key-value pair as a value
+by indenting the nested key.
+
+```yaml
+key:
+  another_key: "another value"
+```
+
+### Multi-line Strings
+
+If the value is a multi-line string,
+use the `|` character,
+followed by any number of lines.
+This is especially useful for lengthy commands.
+
+```yaml
+haiku: |
+  Consider me
+  As one who loved poetry
+  And persimmons.
+```
+
+**Note**:
+Quotes are not necessary
+when using multiline strings.
+
+### Sequences
+
+Keys and values are not restricted to [scalars](https://softwareengineering.stackexchange.com/questions/238033/what-does-it-mean-when-data-is-scalar).
+You can also map a scalar to a sequence.
+
+```yaml
+scalar:
+  - never
+  - gonna
+  - give
+  - you
+  - up
+```
+
+Items in sequences can also be key-value pairs.
+
+```yaml
+simulation:
+  - within: "a simulation"
+  - without:
+      a_glitch: "in the matrix"
+```
+
+**Note**:
+Remember to properly indent a key-value pair
+when it is the value of an item in a sequence.
+
+### Anchors and Aliases
+
+To [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) up your `config.yml`,
+you can use anchors and aliases.
+Anchors are identified by an `&` character,
+and aliases by an `*` character.
+
+```yaml
+song:
+  - &name Al
+  - You
+  - can
+  - call
+  - me
+  - *name
+```
+
+When this list is read by a YAML parser,
+the literal output looks like this.
+
+```yaml
+song:
+  - Al
+  - You
+  - can
+  - call
+  - me
+  - Al
+```
+
+### Merging Maps
+
+Anchors and aliases work for scalar values,
+but to save maps or sequences,
+use `<<` to inject the alias.
+
+```yaml
+default: &default
+  school: hogwarts
+
+harry:
+  <<: *default
+  house: gryffindor
+
+draco:
+  <<: *default
+  house: slytherin
+```
+
+You can also merge multiple maps.
+
+```yaml
+good: &one
+  - harry
+  - ron
+bad: &two
+  - crabbe
+  - goyle
+students:
+  <<: [*one, *two]
+```
+
+For a more complex example,
+see [this gist](https://gist.github.com/bowsersenior/979804).
+
+## See More
+
+While YAML has several other features,
+the examples above should be enough
+to write an robust, functional `config.yml`.
```

Comment:
robust, functional -> to get you started with YAML and help you prevent long, repetitive CircleCI configuration files.

Reply:
👍 to removing "robust, functional"; that's definitely not detailed enough.